### PR TITLE
[network] Add parameter for limiting network interfaces to fix performance issues

### LINKED ISSUE #15437: [network] Pingdevice on Docker-Install results in massive threads
if you are using network-binding with a pingdevice under docker, which normally is used to run not only the container openhab but other containers too you have also a lot of bridge networks which links container. This can result in massive open threads through multiplication. So 30 networks and 10 pingprofiles could impact 300 parallel threads.
On my system load is about 20 if the 10 profiles started to a system-load about 1 if all profiles are disabled.

## Expected Behavior
on docker it should should only start 1 thread from 1 network per ping-profile

## Current Behavior
it opens for 1 ping-profile 1 thread per network instead of 1 thread
- you will get output of all threads loke that:
```
nobody      4261  0.0  0.0   9140  4836 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-e77d9bdcb692 192.168.2.254
nobody      4264  0.5  0.0   9140  4772 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-3a8bcb7e1a20 192.168.2.254
nobody      4270  0.5  0.0   9140  4804 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-5dbd2b0f47cf 192.168.2.254
nobody      4274  0.5  0.0   9140  4876 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-4c02ad546123 192.168.2.254
nobody      4277  0.5  0.0   9140  4884 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-c8c8be0340ae 192.168.2.254
nobody      4283  1.0  0.0   9140  4872 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-d125af02b9b4 192.168.2.254
nobody      4285  0.5  0.0   9140  4828 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-72ac99832aa5 192.168.2.254
nobody      4288  1.0  0.0   9140  4916 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-b3d1fee00881 192.168.2.254
nobody      4290  0.5  0.0   9140  4848 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-77b0d455dbc2 192.168.2.254
nobody      4292  0.5  0.0   9140  4796 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-37b7e1374480 192.168.2.254
nobody      4294  0.5  0.0   9140  4800 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-c6afe6811015 192.168.2.254
nobody      4296  0.5  0.0   9140  4956 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-ad8705eb98cb 192.168.2.254
nobody      4298  1.0  0.0   9140  4892 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-8a8f9f51418c 192.168.2.254
nobody      4309  0.5  0.0   9140  4836 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-eb09dae5f21e 192.168.2.254
nobody      4322  0.5  0.0   9140  4888 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-32b28ae41ba6 192.168.2.254
nobody      4331  0.5  0.0   9140  4880 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-bf53bf1036dd 192.168.2.254
nobody      4334  0.5  0.0   9140  4884 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-f89f1a0e504d 192.168.2.254
nobody      4346  1.0  0.0   9140  4872 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-2db6488b5fc1 192.168.2.254
nobody      4347  1.0  0.0   9140  4936 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-b6dd5302b3c3 192.168.2.254
nobody      4355  1.0  0.0   9140  4884 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-225545b56f9e 192.168.2.254
nobody      4357  1.0  0.0   9140  4880 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-c1e8628f5e78 192.168.2.254
nobody      4360  1.0  0.0   9140  4768 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-ba49e533177b 192.168.2.254
nobody      4373  1.0  0.0   9140  4824 ?        S    08:29   0:00 arping -w 5 -C 1 -i docker0 192.168.2.254
nobody      4374  1.0  0.0   9140  4888 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-0ced1bc112c8 192.168.2.254
nobody      4377  0.0  0.0   9140  4884 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-09b6e1630d76 192.168.2.254
nobody      4380  1.0  0.0   9140  4932 ?        S    08:29   0:00 arping -w 5 -C 1 -i br-46ec93053821 192.168.2.254

```

## Possible Solution
use only the selected network from system-settings-network in openhab, 
OR 
make network-port selectable
OR
let routing of system decide where the package has to go to

## Steps to Reproduce (for Bugs)
- install openhab 4.0.1 (possibly other versions has this too) on docker, network mode host
- select under system settings network your main ethernet of the docker host
- create multiple app-Containers and bridge networks which connecting them, maybe an app-container linking to a database. If you dont know which containers link apache{1,2,3} to mysql{1,2,3} to get 3 apps with their database and their own bridge between)
- verify that you have multiple bridges created with `docker network ls`
- create a network-binding profile like that:
```
UID: network:pingdevice:4b7f079f3f
label: InternetRouter
thingTypeUID: network:pingdevice
configuration:
  hostname: 192.168.2.254
  refreshInterval: 60000
  retry: 1
  timeout: 5000
```
- enable this profile and directly then directly watch next command
- `docker exec  [containername] ps aux|grep arping`
(repeat if you want until next refreshinterval if you want)

## Context
for 1 ping on my side 26 arping-threads are opened which resulting on pinging multiple devices in a high thread. I would expect 1 thread per ping-profile
Also created Community-Discussions: [https://community.openhab.org/t/openhab-4-0-1-and-network-pingdevice-on-docker-install-results-in-massive-threads/148658](https://community.openhab.org/t/openhab-4-0-1-and-network-pingdevice-on-docker-install-results-in-massive-threads/148658)

## Your Environment

* Version used: openhab/openhab:4.0.1 on docker
* Operating System and version : Debian GNU/Linux 11 (bullseye), Docker-CE

### PR DESCRIPTION
**Pull request by wborn**
* Add configuration parameter for limiting network interfaces
* Use last known reachable interface

Fixes #15437

Depends on https://github.com/openhab/openhab-core/pull/3981

### CODE REVIEW

File: bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -54,6 +54,14 @@
 				<default>5000</default>
 				<advanced>true</advanced>
 			</parameter>
+
+			<parameter name="networkInterfaceNames" type="text" required="false" multiple="true">
```
Comment (lolodomo) [inline bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml:58]: Please also update README to add the description of the two new parameters.

**Review by lolodomo — CHANGES_REQUESTED** [review]
Code is good, just an update of documentation is missing

File: bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml (hunk 1 above)
Comment (wborn) [reply bundles/org.openhab.binding.network/src/main/resources/OH-INF/thing/thing-types.xml:58]: I have added a commit for this!

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by kaikreuzer** [comment]
@wborn Is this something that should be backported to 4.1.x?

**Comment by wborn** [comment]
It would be nice to get this solved. Though https://github.com/openhab/openhab-webui/pull/2246 is also needed for proper UI support.

**Comment by kaikreuzer** [comment]
@florian-h05 Do you think https://github.com/openhab/openhab-webui/pull/2246 can be merged and also backported tonight?

**Comment by florian-h05** [comment]
I planned to do that before leaving home, but didn’t have the time.
I will be late at home tonight, but I can try to do that ASAP after getting back home.

**Comment by kaikreuzer** [comment]
Great, thank you. It is a minimal code change - if you say, it looks alright, I can also merge and cherry-pick, if you like.

**Comment by kaikreuzer** [comment]
Cherry-picked to `4.1.x`.

**Pull request merged**